tests/test_class_contracts.py

```diff
@@ -67,6 +72,16 @@ def test_change_name_invalid() -> None:
     msg = str(excinfo.value)
     assert 'len(self.name) > 0' in msg
 
+def test_change_name_invalid_nonalpha() -> None:
+    """
+    Change name to contain non-alphabet characters. Expect an exception.
+    """
+    with pytest.raises(AssertionError) as excinfo:
+        p1.change_name("David")  # Changing it back to a valid value.
```

Comment:
You might have added this line because the test was failing? It would be great if you explored [pytest fixtures](https://docs.pytest.org/en/latest/fixture.html#fixtures-as-function-arguments) as a way to remove the global variable in your tests (same as the new tests down below, too).

Reply:
I'll do this for my next PR in which I focus on rewriting tests and adding more complex ones